angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
I am heavily leaning in favor of the following overloads
```ts
<A extends null | undefined | null[] | undefined[]>(array: A, expression?: string | string[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): A;
<T>(array: T[], expression?: string | ((value: T) => any) | (((value: T) => any) | string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```
and the addition of this test
```ts
function testOrderByFilter() {
    const ordered = angular.injector(['ng']).get<angular.IFilterOrderBy>('orderByFilter')([1, 2, 3], x => x); // number[]
    const ordered1 = angular.injector(['ng']).get<angular.IFilterOrderBy>('orderByFilter')([null, null], x => x); // null[]
    const orderedNull = angular.injector(['ng']).get<angular.IFilterOrderBy>('orderByFilter')(null); // null
    const orderedUndefined = angular.injector(['ng']).get<angular.IFilterOrderBy>('orderByFilter')([undefined, undefined], x => x); // undefined[]
}
```